### PR TITLE
Set WebApplication.New().ServerFeatures

### LINKED ISSUE #40042: WebApplication.New().ServerFeatures is null
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When using minimal APIs in ASP.NET Core 6, after creating a new application from the existing app, the ServerFeatures property of the new application is null. 

Per @Tratcher, this is because the ServerFeatures are never put in the property bag of the application: https://github.com/dotnet/aspnetcore/blob/d1c349f29752ac0718bda7b809ab7fd4f82ca414/src/DefaultBuilder/src/WebApplication.cs#L70-L71.

### Expected Behavior

ServerFeatures shouldn't be null.

### Steps To Reproduce

```c#
var builder = WebApplication.CreateBuilder(args);
var app = builder.Build();
var newApp = (app as IApplicationBuilder).New();
Debug.Assert(newApp.ServerFeatures != null);
```

### Exceptions (if any)

_No response_

### .NET Version

6.0.100

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by feiyun0112**
fix #40042

### CODE REVIEW

File: src/DefaultBuilder/src/WebApplication.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public sealed class WebApplication : IHost, IApplicationBuilder, IEndpointRouteB
     internal WebApplication(IHost host)
     {
         _host = host;
-        ApplicationBuilder = new ApplicationBuilder(host.Services);
+        ApplicationBuilder = new ApplicationBuilder(host.Services, this.ServerFeatures);
```
Comment (Tratcher) [inline src/DefaultBuilder/src/WebApplication.cs:30]: Style: Avoid `this.` unless it's required. You don't need it here do you?
```suggestion
        ApplicationBuilder = new ApplicationBuilder(host.Services, ServerFeatures);
```

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
@pranavkm @HaoK why would this build fail with identity nullability issues?

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Closing and re-opening to fix the cla check.

**Comment by campersau** [comment]
Are you considering backporting this to 6.0 ?

**Comment by Tratcher** [comment]
> Are you considering backporting this to 6.0 ?

Unlikely, given there's a workaround. Is this impacting you on 6.0?

**Comment by Tratcher** [comment]
Thanks @feiyun0112

**Pull request merged**